__tests__/git.test.ts

```diff
@@ -162,3 +162,32 @@ test('diff returns contextless diff', async () => {
 +++ b/file.txt
 @@ -1,0 +2 @@ ${initialContent}+${addedContent}`);
 });
+
+test(`diff can handle HEAD~ as commitish argument`, async () => {
+  // arrange
+  const tmpdir = await createTempdir();
+  const filename = 'file.txt';
+  const filepath = path.join(tmpdir, filename);
+  const initialContent = `Hello, there!
+`; // note that this trailing newline is important for git to consider the edit as a pure addition
```

Comment:
Not sure about this but I think this could create two additions instead. Inserting only `Hello, there!` would be just one pure addition.

Reply:
Oh, no that's the initial content. It's when _adding to that_ that there's trouble. If you don't add that trailing newline, then on inserting the modified content (on the next line), Git sees it as a deletion of `Hello, there!` and an insertion of `Hello, there!\n`.

That is to say, without having the newline there initially, the diff is this:

```diff
-Hello, there!
+Hello, there!
+what is up?
```
But if we take care to tack on that newline right from the get-go, we get this diff instead:
```diff
+what is up?
```

Reply:
Thanks for the explanation. This logic is analogous to why trailing commas are added so it makes sense to me now 😅.